pandas/tests/frame/indexing/test_setitem.py

```diff
@@ -298,6 +299,36 @@ def test_iloc_setitem_bool_indexer(self, klass):
         expected = DataFrame({"flag": ["x", "y", "z"], "value": [2, 3, 4]})
         tm.assert_frame_equal(df, expected)
 
+    def test_setitem_complete_columns_different_dtypes(self):
```

Comment:
can you add an int64 column in the original frame (that is not selected).

also can you test with Int64.

Reply:
Added the column.

It seems like astype does not work for ``Int64`` if the input if from object. Should I add a completly new test or is there a trick I am not aware of?

Reply:
yeah that's right, you can `.astype('int64').astype(dtype)`

Reply:
Thanks very much, that is pretty obvious. Parametrized the test now